src/Services/InsuranceService.php

```diff
@@ -60,25 +83,100 @@ public function validate($data)
         return $validator->validate($data);
     }
 
-    public function getOne($pid, $type)
+    public function getOne($uuid)
```

Comment:
This won't work. This class is used in several places in core. Our services, while used in our apis, are still used by core code.
e.g interface/reports/receipts_by_method_report.php

We are not ready to institute UUID throughout all of core for services and should stay with pid and table ids etc. as that is how they are referred elsewhere.
Also, any new services need to keep this in mind.

Reply:
as an addendum to my comment you could pass in uuid and arbitrate if an int i.e table id or a string i.e. uuid then handle accordingly. Still leave 'type' for standard lookups.

Reply:
modified as per your suggestion, also addressed similarly for getAll()